### DISCUSSION TITLE
Implementing an additional option to retrieve all properties except specific ones

### DISCUSSION THREAD

**Discussion by AbdlrahmanSaberAbdo** [mikro-orm/mikro-orm#4460] [Ideas]
Referring to the discussion found at https://github.com/mikro-orm/mikro-orm/discussions/2984, I would like to introduce a new functionality that, to the best of my knowledge, has not been implemented yet.

My idea is to create an option called `allFieldsWithout` or `fieldsWithout`. To get started with the code base, I examined how the existing `fields` option is implemented. I noticed that the `buildFields` method is used in the following classes:

- EntityManager
- EntityLoader
- AbstractSQLDriver
- MongoDriver

To begin implementing this option, I plan to first add it to the `findOptions` interface. Now, my question is whether I should implement the `exclude` method in each of these classes. My approach would involve excluding the specified property from the metadata's `props` and then including all other properties. Should this functionality also support excluding fields from relations? Initially, I'd like to focus on excluding props specific to the specified entity/table.

Additionally, I believe it's necessary to impose some restrictions to prevent the user from specifying both options (`fields` and `fieldsWithout`) simultaneously. What are your thoughts on this?

Before proceeding, I'd like to make sure that you're welcome with this option, @B4nan. It would also be greatly appreciated if you could provide some guidance to help me get started.

**Comment by B4nan** [comment]
I'd call it just exclude.
